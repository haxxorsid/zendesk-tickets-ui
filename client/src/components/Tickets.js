import React from 'react';
import Ticket from './Ticket';
import Paginator from './Paginator.js';

function Tickets(props) {
    return (
        <div className="tickets">
            <h3>{props.count} total tickets, {props.countPerPage} on this page</h3>
            {props.tickets &&
                props.tickets.map(ticket => <Ticket key={ticket.id} ticket={ticket} onSelectTicket={props.onSelectTicket} />)
            }
            <Paginator
                onPageChange={props.onPageChange}
                numberOfPages={props.numberOfPages}
                currentPage={props.currentPage}
            />
        </div>

    );
}

export default Tickets;