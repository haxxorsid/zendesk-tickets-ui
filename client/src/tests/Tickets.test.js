import React from 'react';
import renderer from 'react-test-renderer'
import Tickets from '../components/Tickets';
import ticketsData from './mockData/tickets'

test('Should render tickets correctly with props', () => {
    const component = renderer
        .create(<Tickets tickets={ticketsData} count={ticketsData.length} countPerPage={2} />);
    expect(component.toJSON()).toMatchSnapshot()
  })
  
  test('Should render ticket title correctly', () => {
    const component = renderer.create(<Tickets />);
    expect(component.root.findAllByType('h3').length).toBe(1)
  })
  
  test('Should render paginator correctly', () => {
    const component = renderer.create(<Tickets />);
    expect(component.root.findAllByProps({className: 'paginator'}).length).toBe(1)
  })