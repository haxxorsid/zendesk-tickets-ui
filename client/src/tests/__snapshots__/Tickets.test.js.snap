// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render tickets correctly with props 1`] = `
<div
  className="tickets"
>
  <h3>
    4
     total tickets, 
    2
     on this page
  </h3>
  <div
    className="ticket"
    onClick={[Function]}
  >
    <div
      className="ticket-header"
    >
      <p>
        <span
          className="id"
        >
           ID: 
          1
           
        </span>
        <span
          className="status"
        >
          Status: 
        </span>
        <span
          className="right date"
        >
          Invalid Date
        </span>
      </p>
      <p>
        <b>
          Subject: 
        </b>
         
        <span
          className="subject"
        />
      </p>
    </div>
  </div>
  <div
    className="ticket"
    onClick={[Function]}
  >
    <div
      className="ticket-header"
    >
      <p>
        <span
          className="id"
        >
           ID: 
          2
           
        </span>
        <span
          className="status"
        >
          Status: 
        </span>
        <span
          className="right date"
        >
          Invalid Date
        </span>
      </p>
      <p>
        <b>
          Subject: 
        </b>
         
        <span
          className="subject"
        />
      </p>
    </div>
  </div>
  <div
    className="ticket"
    onClick={[Function]}
  >
    <div
      className="ticket-header"
    >
      <p>
        <span
          className="id"
        >
           ID: 
          3
           
        </span>
        <span
          className="status"
        >
          Status: 
        </span>
        <span
          className="right date"
        >
          Invalid Date
        </span>
      </p>
      <p>
        <b>
          Subject: 
        </b>
         
        <span
          className="subject"
        />
      </p>
    </div>
  </div>
  <div
    className="ticket"
    onClick={[Function]}
  >
    <div
      className="ticket-header"
    >
      <p>
        <span
          className="id"
        >
           ID: 
          4
           
        </span>
        <span
          className="status"
        >
          Status: 
        </span>
        <span
          className="right date"
        >
          Invalid Date
        </span>
      </p>
      <p>
        <b>
          Subject: 
        </b>
         
        <span
          className="subject"
        />
      </p>
    </div>
  </div>
  <ul
    className="paginator"
  />
</div>
`;
